sw/device/silicon_creator/rom_ext/rom_ext_start.S

```diff
@@ -147,18 +151,126 @@ _rom_ext_start_boot:
    * If an exception fires, the handler is conventionaly only allowed to clobber
    * memory at addresses below `sp`.
    */
-  la   sp, (_stack_end - 16)
+  
+  lui sp, %hi(_stack_end - 16)
+  addi sp, sp, %lo(_stack_end - 16)
+  
+  /**
+   * Initialize the Ibex simple address translation to run position dependent
+   * code in the virtual flash space.
+   */
+  /**
+   * Compute the current slot address by: 
+   * `addr = pc - _addr_translation_crt_offset`
+   */
+  lui t0, %hi(_addr_translation_crt_offset)
+  addi t0, t0, %lo(_addr_translation_crt_offset)
+  
+  sub a0, a0, t0 // First address physical.
+  // mask = virtual_addr | ((size - 1) >> 1);
+  lui a1, %hi(_vflash_address)
+  addi a1, a1, %lo(_vflash_address)
+  lui t0, %hi(_vflash_size - 1)
+  addi t0, t0, %lo(_vflash_size - 1)
+  srli t0, t0, 1
+  or a1, a1,t0 
+
+  // Load ibex config base address.
+  li t0, TOP_EARLGREY_RV_CORE_IBEX_CFG_BASE_ADDR 
+
+  // Set ibus and dbus address matching.
+  sw a1, RV_CORE_IBEX_IBUS_ADDR_MATCHING_0_REG_OFFSET(t0)
+  sw a1, RV_CORE_IBEX_DBUS_ADDR_MATCHING_0_REG_OFFSET(t0)
+
+  // Set ibus and dbus address remap.
+  sw a0, RV_CORE_IBEX_IBUS_REMAP_ADDR_0_REG_OFFSET(t0)
+  sw a0, RV_CORE_IBEX_DBUS_REMAP_ADDR_0_REG_OFFSET(t0)
+
+  // Enable address translation.
+  li t1, 0x01
+  sw t1, RV_CORE_IBEX_IBUS_ADDR_EN_0_REG_OFFSET(t0)
+  sw t1, RV_CORE_IBEX_DBUS_ADDR_EN_0_REG_OFFSET(t0) 
 
   /**
-   * Set well-defined interrupt/exception handlers
-   *
-   * The lowest two bits should be `0b01` to ensure we use vectored interrupts.
+   * The registers pmpaddr7, pmpaddr8 and pmpaddr9 shall be all zeros at this point.
+   *  ``` 
+   *  if ((pmpaddr7 | pmpaddr8 | pmpaddr9) != 0){
+   *     asm("unimp");
+   *  }
+   *  ``` 
    */
-  la   t0, (_rom_ext_interrupt_vector + 1)
-  csrw mtvec, t0
+  csrr t1, pmpaddr7
+  csrr t0, pmpaddr8
+  or t1, t1, t0
+  csrr t0, pmpaddr9
+  or t1, t1, t0
+  beqz t1, _rom_ext_start_boot_set_epmp_
```

Comment:
Since this isn't a hardening feature (at least, not yet!) I'd reverse this jump, so:

```
  bnez t1, _rom_ext_epmp_error
...

_rom_ext_epmp_error:
  unimp
```

That way the happy path is kept as the linear path.

Reply:
Done